### PR TITLE
MTM-45799 add generic information about inventory roles improvements …

### PR DESCRIPTION
**Pull request by jkot-softwareag**
…for 10.13 version

### CODE REVIEW

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 1]
```diff
@@ -377,6 +377,17 @@ If you try to perform actions without sufficient permissions, an error message w
...
+### Performance improvements
+
+Starting with version 10.13 of the {{< product-c8y-iot >}} platform, several improvements have been implemented for users with inventory roles access.
+In earlier versions responses were slow for tenants with large inventory hierarchies.
+The following UI pages have an improved performance after updating to version {{< c8y-current-version >}}:
+* In the [device details view](/users-guide/device-management/#device-details), the tabs **Info**, **Measurements**, **Alarms**, **Events** and **Control**.
+* Pages with aggregated alarm views from multiple devices, if the number of alarms in the system is low, for example, [Cockpit > Home dashboard](/users-guide/cockpit/#home-dashboard), Cockpit > Alarms and Device Management > Home.
+* Pages with aggregated events and operations from multiple devices, for example, Device Management > Overviews > Events 
+
+New approach is disabled by default and needs to be switched on the tenant level using tenant option (see the Release notes for details).
```
Comment (jkot-softwareag) [inline content/users-guide/administration-bundle/managing-permissions.md:389]: or maybe I should change last part to something which I found in this repo:
```(see the [Release notes](https://{{< domain-c8y >}}/releasenotes/release-{{< c8y-current-version-hyphen >}}/overview-{{< c8y-current-version-hyphen >}}) for details)
```

**Review by pawelkowalczyksag — APPROVED** [review]

**Review by d-mariak — APPROVED** [review]

File: content/users-guide/administration-bundle/managing-permissions.md (hunk 1 above)
Comment (rssag) [reply content/users-guide/administration-bundle/managing-permissions.md:389]:
```suggestion
The improvements are disabled by default in version {{< c8y-current-version >}}. Activate them on tenant level by using the tenant option. See the relevant [release notes entry](https://cumulocity.com/releasenotes/release-10-13-0/whatsnew-10-13-0/#inventory-roles-performance-improvement) for details.
```

**Review by rssag — APPROVED** [review]

**Comment by rssag** [comment]
@BeateRixen

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 2]
```diff
@@ -377,6 +377,17 @@ If you try to perform actions without sufficient permissions, an error message w
 
 To help troubleshooting permissions, click the **User** button at the right of the top bar. From the context menu, select **Access denied requests**. In the resulting window details on the denied accesses are provided. An administrator user or the [product support](/welcome/contacting-support/) can help in fixing the permissions.
 
+### Performance improvements
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:380]: @jkot-softwareag I don´t think that is makes sense to add this section to the documentation. Stating performance improvements in the release notes is the right place. It is of no value for users here.

Comment (BeateRixen) [reply content/users-guide/administration-bundle/managing-permissions.md:380]: Or we should describe this as a new feature which allows to activate options on tenant level which improve the performance. But shouldn´t this then go into the operations documentation?

Comment (BeateRixen) [reply content/users-guide/administration-bundle/managing-permissions.md:380]: Could be something like:

### Optimizing the performance

You can optimize the performance for tenants with large inventory hierarchies by activating the respective tenant option (name of option, link to operations documentation?) which is disabled by default. 

The following UI pages will show an improved performance:
* In the [device details view](/users-guide/device-management/#device-details), the tabs **Info**, **Measurements**, **Alarms**, **Events**.
* Pages with aggregated alarm views from multiple devices, if the number of alarms in the system is low, for example, [Cockpit > Home dashboard](/users-guide/cockpit/#home-dashboard), Cockpit > Alarms and Device Management > Home.
* Pages with aggregated events from multiple devices, if the number of events is low, for example, [Device Management > Overviews > Events](/users-guide/device-management/#to-view-events).

A link to the release notes is not required here. Also, versions do not need to be added, as this description only shows up in the relevant versions.

Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:380]:
```suggestion
### Improving the performance
```

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 3]
```diff
@@ -377,6 +377,17 @@ If you try to perform actions without sufficient permissions, an error message w
...
+
+Starting with version 10.13 of the {{< product-c8y-iot >}} platform, several improvements have been implemented for users with inventory roles access.
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:382]:
```suggestion
As an administrator, you can optimize the UI performance for users with inventory roles access and particularly speed up requests for tenants with large inventory hierarchies.  

This performance feature can be enabled:

-  on platform level via the configuration file (see the *{{< product-c8y-iot >}} - Operations guide* for details).

- on tenant level via a tenant option. The tenant option has 2 possible values: LEGACY/OPTIMIZED, where LEGACY is the global default.

The option looks like the following in the REST API:

{ "category": "configuration", "key": "acl.algorithm-version", "value": "OPTIMIZED"}
```

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 4]
```diff
@@ -377,6 +377,17 @@ If you try to perform actions without sufficient permissions, an error message w
...
+In earlier versions responses were slow for tenants with large inventory hierarchies.
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:383]:
```suggestion

```

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 5]
```diff
@@ -377,6 +377,17 @@ If you try to perform actions without sufficient permissions, an error message w
...
+The following UI pages have an improved performance after updating to version {{< c8y-current-version >}}:
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:384]:
```suggestion
The performance of the following UI pages can be improved by this option:
```

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 6]
```diff
@@ -377,6 +377,17 @@ If you try to perform actions without sufficient permissions, an error message w
...
+* In the [device details view](/users-guide/device-management/#device-details), the tabs **Info**, **Measurements**, **Alarms**, **Events**.
+* Pages with aggregated alarm views from multiple devices, if the number of alarms in the system is low, for example, [Cockpit > Home dashboard](/users-guide/cockpit/#home-dashboard), Cockpit > Alarms and Device Management > Home.
+* Pages with aggregated events from multiple devices, if the number of events is low, for example, [Device Management > Overviews > Events](/users-guide/device-management/#to-view-events).
+
+The improvements are disabled by default in version {{< c8y-current-version >}}. Activate them on tenant level by using the tenant option. See the relevant [release notes entry](https://cumulocity.com/releasenotes/release-10-13-0/whatsnew-10-13-0/#inventory-roles-performance-improvement) for details.
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:389]:
```suggestion
By default, this option is disabled. 
```

**Comment by jkot-softwareag** [comment]
@BeateRixen this is a screenshot how it locally looks now (I have added ` to have monospaced font for tenant option, since this is "json code")

image

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 7]
```diff
@@ -377,6 +377,25 @@ If you try to perform actions without sufficient permissions, an error message w
 
 To help troubleshooting permissions, click the **User** button at the right of the top bar. From the context menu, select **Access denied requests**. In the resulting window details on the denied accesses are provided. An administrator user or the [product support](/welcome/contacting-support/) can help in fixing the permissions.
 
+### Improving the performance
+
+As an administrator, you can optimize the UI performance for users with inventory roles access and particularly speed up requests for tenants with large inventory hierarchies.
+
+This performance feature can be enabled:
+- on platform level via the configuration file (see the *{{< product-c8y-iot >}} - Operations guide* for details).
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:385]:
```suggestion
- on platform level via the configuration file (only available for platform administrators, see the *{{< product-c8y-iot >}} - Operations guide* for details).
```

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 8]
```diff
@@ -377,6 +377,25 @@ If you try to perform actions without sufficient permissions, an error message w
...
+
+This performance feature can be enabled:
+- on platform level via the configuration file (only available for platform administrators, see the *{{< product-c8y-iot >}} - Operations guide* for details).
+- on tenant level via a tenant option. The tenant option has 2 possible values: LEGACY/OPTIMIZED, where LEGACY is the global default.
+  
+The option looks like the following in the REST API: 
+
+`{"category": "configuration", "key": "acl.algorithm-version", "value": "OPTIMIZED"}`
+
+The performance of the following UI pages can be improved by this option:
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:392]:
```suggestion
The performance of the following UI pages will be improved when this option is enabled:
```

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 9]
```diff
@@ -377,6 +377,25 @@ If you try to perform actions without sufficient permissions, an error message w
...
 
+### Improving the performance
+
+As an administrator, you can optimize the UI performance for users with inventory roles access and particularly speed up requests for tenants with large inventory hierarchies.
+
+This performance feature can be enabled:
+- on platform level via the configuration file (only available for platform administrators, see the *{{< product-c8y-iot >}} - Operations guide* for details).
+- on tenant level via a tenant option. The tenant option has 2 possible values: LEGACY/OPTIMIZED, where LEGACY is the global default.
+  
+The option looks like the following in the REST API:
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:388]:
```suggestion
The option looks like the following in the REST API (see also the [{{< openapi >}}](https://cumulocity.com/api/10.13.0/#operation/postOptionCollectionResource): 
```

File: content/users-guide/administration-bundle/managing-permissions.md [hunk 10]
```diff
@@ -377,6 +377,25 @@ If you try to perform actions without sufficient permissions, an error message w
...
+As an administrator, you can optimize the UI performance for users with inventory roles access and particularly speed up requests for tenants with large inventory hierarchies.
+
+This performance feature can be enabled:
+- on platform level via the configuration file (only available for platform administrators, see the *{{< product-c8y-iot >}} - Operations guide* for details).
+- on tenant level via a tenant option. The tenant option has 2 possible values: LEGACY/OPTIMIZED, where LEGACY is the global default.
+  
+The option looks like the following in the REST API: 
+
+`{"category": "configuration", "key": "acl.algorithm-version", "value": "OPTIMIZED"}`
+
```
Comment (BeateRixen) [inline content/users-guide/administration-bundle/managing-permissions.md:391]:
```suggestion

The setting on tenant level has priority over the setting on platform level.
```

**Comment by BeateRixen** [comment]
@jkot-softwareag I have added one more sentence that has somehow slipped through ("The setting on tenant level has priority over the setting on platform level."). And I have added the link to the API documentation. Please check again.

If approved, we could graft this to 10.14 and 10.15. Only change to be done is the link to the openapi, which at the moment still needs a version in it as without version it currently points to the landing page.

For 10.16 we would only additionally have to change the default setting.

**Comment by jkot-softwareag** [comment]
Screenshot from current version:

image

**Pull request merged**